### DISCUSSION TITLE
Serverless deployment with google cloud storage

### DISCUSSION THREAD

**Discussion by MS-Kinguin** [langflow-ai/langflow#1519] [General]
Hello,

I'm currently exploring the capabilities of LangFlow and am impressed by what it can do. I have a question regarding its potential deployment for production use. Specifically, I'm curious whether it's feasible to deploy a fully functional LangFlow application using a Docker image, perhaps hosted on Google Cloud Run, for production purposes. I've already conducted a preliminary test and found that it functions as follows:

```
FROM python:3.9-slim

WORKDIR /app

# Install dependencies
RUN apt-get update && apt-get install -y \
    build-essential \
    cmake \
    git \
    && rm -rf /var/lib/apt/lists/*

# Install LangFlow
RUN pip install --no-cache-dir langflow

# Copy local code to the container
COPY . .

# Set the port the app runs on
EXPOSE 8080

# Set environment variables
ENV LANGFLOW_HOST=0.0.0.0
ENV LANGFLOW_PORT=8080
ENV LANGFLOW_LOG_LEVEL=info

# Run the application
CMD ["langflow", "run"]
```

I've noticed a potential issue with this configuration: it temporarily stores flows in memory, which may not be ideal for production. I'm considering the use of cloud storage so that the flows can be shared more effectively. Additionally, I have concerns regarding the integration of LangFlow's API querying within a serverless environment for production. Could you provide some insights on whether LangFlow can be efficiently utilized in such an environment?

**Comment by MS-Kinguin** [comment]
@dosu-bot

**Reply by codenprogressive** [reply]
Hi @MS-Kinguin, were you able to implement a postgresSQL (gcp cloud SQL) as an external db? Also, how are you managing authentication? Any OAuth integration?

**Reply by vktinker** [reply]
Hey @MS-Kinguin Have you got this now working in production?? I am at the point of deciding this for production vs typical vm+ managed instance group + load balancer
